gax-java/gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -98,8 +91,28 @@ public static String getGaxVersion() {
     return GAX_VERSION;
   }
 
-  /** Returns the current runtime version */
-  private static String getRuntimeVersion() {
-    return System.getProperty("java.version");
+  /**
+   * Returns the current runtime version. For GraalVM the values in this method will be fetched at
+   * build time and the values should not differ from runtime (executable)
+   */
+  @VisibleForTesting
+  static String getRuntimeVersion() {
+    String javaRuntimeInformation = System.getProperty("java.version");
```

Comment:
Can `java.version` be `null`?

Reply:
I think there is a possibility of this happening. In such a case, would we still want to capture the vendor information, since the vendor version might help us deduce java-version?

Reply:
In such a case, I want this method not to fail with NullPointerException. @ddixit14  Can you add a test case for it?